mlflow/pipelines/steps/ingest/__init__.py

```diff
@@ -100,6 +101,12 @@ def _run(self, output_directory: str) -> BaseCard:
         )
         return step_card
 
+    def _validate_user_code_output(self, ingested_df):
+        if not isinstance(ingested_df, pd.DataFrame):
+            raise MlflowException(
+                message="The `ingested_data` is not a DataFrame, please ingest the correct data format"
```

Comment:
This message is not helpful enough. As a user I'd have a hard time to figure out what went wrong. It should be directly connected to the user defined function name and its required output.